app/validators/termlist_has_no_duplicates.rb

```diff
@@ -0,0 +1,21 @@
+# Verifies that all terms in a list are unique within the list
+class TermlistHasNoDuplicates < ActiveModel::Validator
+  def validate(record)
+    @uris_seen = Hash.new(0)
+    record.terms.each do |term|
```

Comment:
So, there's some micro-optimization here that may or may not be useful, but clearer is `term_subjects.uniq.length != term_subjects.length`. Something to consider?